src/main/java/com/mineboundteam/minebound/entity/RockProjectile.java

```diff
@@ -0,0 +1,110 @@
+package com.mineboundteam.minebound.entity;
+
+import net.minecraft.core.particles.ParticleTypes;
+import net.minecraft.sounds.SoundEvents;
+import net.minecraft.world.damagesource.DamageSource;
+import net.minecraft.world.entity.Entity;
+import net.minecraft.world.entity.EntityType;
+import net.minecraft.world.entity.LivingEntity;
+import net.minecraft.world.entity.Mob;
+import net.minecraft.world.entity.projectile.AbstractArrow;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.phys.BlockHitResult;
+import net.minecraft.world.phys.EntityHitResult;
+import net.minecraft.world.phys.Vec3;
+
+
+import java.util.List;
+
+import static com.mineboundteam.minebound.entity.registry.EntityRegistry.ROCK_PROJECTILE;
+
+
+/**
+ * Derived from {@link FireProjectile}
+ * Derived from {@link net.minecraft.world.entity.projectile.AbstractArrow}
+ */
+public class RockProjectile extends AbstractArrow {
+    private float damage;
+    private float damageRadius;
+    public RockProjectile(EntityType<? extends RockProjectile> pEntityType, Level pLevel) {
+        super(pEntityType, pLevel);
+    }
+
+    public RockProjectile(Level pLevel, LivingEntity shooter, double pX, double pY, double pZ, float damage, float damageRadius) {
+        super(ROCK_PROJECTILE.get(), pX, pY, pZ, pLevel);
+        this.setOwner(shooter);
+
+        this.damage = damage;
+        this.damageRadius = damageRadius;
+        this.setSoundEvent(SoundEvents.STONE_BREAK);
+        //Overwrite damage that would have been applied in AbstractArrow
+        this.setBaseDamage(0.0D);
+    }
+
+    @Override
+    protected void onHitEntity(EntityHitResult pResult) {
+        super.onHitEntity(pResult);
+        if(!this.level.isClientSide){
+            Entity player = this.getOwner();
+            List<Entity> hitEntities = this.level.getEntities(this, this.getBoundingBox().inflate(damageRadius));
+            for (Entity target : hitEntities) {
+                if (target instanceof LivingEntity) {
+                    target.hurt(DamageSource.thrown(this, player), damage);
+                    if (player instanceof LivingEntity p) {
+                        this.doEnchantDamageEffects(p, target);
+                    }
+                }
+            }
+
+        }
+    }
+
+    @Override
+    protected void onHitBlock(BlockHitResult pResult) {
+        super.onHitBlock(pResult);
+        if (!this.level.isClientSide) {
+            Entity player = this.getOwner();
+            //Mobs won't explode everything you own
+            if (!(player instanceof Mob) || net.minecraftforge.event.ForgeEventFactory.getMobGriefingEvent(this.level, this)) {
```

Comment:
You're no longer doing `.explode`, so there shouldn't be a reason to do this check anymore.